api/envoy/service/status/v2/csds.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+package envoy.service.status.v2;
+
+import "envoy/admin/v2alpha/config_dump.proto";
+import "envoy/api/v2/core/base.proto";
+import "envoy/type/matcher/metadata.proto";
+import "envoy/type/matcher/string.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+option java_package = "io.envoyproxy.envoy.service.status.v2";
+option java_outer_classname = "CsdsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+
+// CSDS is Client Status Discovery Service. It can be used to get the status of
+// an xDS-compliant client from the management server's point of view. In the
+// future, it can potentially be used as an interface to get the current
+// state directly from the client.
+service ClientStatusDiscoveryService {
+  rpc StreamClientStatus(stream ClientStatusRequest) returns (stream ClientStatusResponse) {
+  }
+
+  rpc FetchClientStatus(ClientStatusRequest) returns (ClientStatusResponse) {
+    option (google.api.http).post = "/v2/discovery:client_status";
+    option (google.api.http).body = "*";
+  }
+}
+
+// Status of a config.
+enum ConfigStatus {
+  // Status info is not available/unknown.
+  UNKNOWN = 0;
+
+  // Management server has sent the config to client and received ACK.
+  SYNCED = 1;
+
+  // Config is not sent.
+  NOT_SENT = 2;
+
+  // Management server has sent the config to client but hasn’t received
+  // ACK/NACK.
+  STALE = 3;
+
+  // Management server has sent the config to client but received NACK.
+  ERROR = 4;
+}
+
+// Request for client status of clients identified by a list of NodeMatchers.
+message ClientStatusRequest {
+  message NodeMatcher {
+    // Specifies match criteria on the client node id.
+    type.matcher.StringMatcher node_id = 1;
+
+    // Specifies match criteria on the client node metadata.
+    type.matcher.MetadataMatcher node_metadata = 2;
+  }
```

Comment:
Can you go ahead and move this out into the types path as an independent matcher, along with the other matchers? I know of another specific use case in which we are going to need to do node matching, and I'm sure there are others that will come up soon. cc @jessicayuen for XDS aggregator overlap on the node matching config.

Reply:
Done.